File: metarealm-growth-os-main/frontend/types/knowledge.ts
/** Knowledge base domain — mirrors backend/app/schemas/knowledge.py. */

export interface KnowledgeDocument {
  id: string;
  title: string;
  filename: string;
  kind: string;
  chunkCount: number;
  embedded: boolean;
}

export interface KnowledgePassage {
  documentTitle: string;
  page?: number | null;
  text: string;
  score: number;
}

export interface AskResult {
  answer: string | null;
  usedLlm: boolean;
  passages: KnowledgePassage[];
}

export interface KnowledgeSyncResult {
  ingested: string[];
  reindexed: string[];
  embeddedBackfilled: string[];
  skipped: number;
  embeddingsAvailable: boolean;
}

File: reference/metarealm-growth-os-main/backend/app/schemas/knowledge.py
from app.schemas.base import CamelModel


class KnowledgeDocumentOut(CamelModel):
    id: str
    title: str
    filename: str
    kind: str
    chunk_count: int
    embedded: bool


class PassageOut(CamelModel):
    document_title: str
    page: int | None = None
    text: str
    score: float


class AskRequest(CamelModel):
    question: str


class AskResponse(CamelModel):
    answer: str | None = None
    used_llm: bool
    passages: list[PassageOut]


class SyncResponse(CamelModel):
    ingested: list[str]
    reindexed: list[str]
    embedded_backfilled: list[str]
    skipped: int
    failed: list[str] = []
    embeddings_available: bool
